docs: Improve README.dev

dougsland opened the pull request:
Trivial changes to README.dev:
     - Add Table of Content in README.dev. 
 
Helps to link specific part of the documentation in a commit or a discussion.

dougsland committed:
build: make build-rpm.sh return correctly

Currently buid-scripts/build-rpm.sh is failing
with mkdir error in the end.

GH-Issue: https://github.com/containers/hirte/issues/282
Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
README.dev: add Table of Content

Make easy for reference in commit or discussions.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

engelmi reviewed (approved):
LGTM

mwperina commented on README.developer.md:
```diff
@@ -153,14 +170,17 @@ After executing a `meson install` the MAN pages are located in the `man/man*` di
 
-Hirte is packaged as an RPM. The scripts used to create the SRPM and RPM may be found in [.build-scripts](./build-scripts).
+Hirte is packaged as an RPM. For a quick RPM creation, use `make rpm` target. This target will call all required
+scripts to generate RPMs and SRPM.
```
I'm not sure about promoting make here, we should really aim for meson as the main build tool and that's why we have those build scripts, which are easy to use.

engelmi replied:
I prepared a small PR to remove the meson dependency from the rpm build scripts:
https://github.com/containers/hirte/pull/307
This way, the make target points to the meson target and it should be fine to promote make here, I think.

mwperina replied:
Personally I don't like advertising the usage of make in the project. We have agreed to use meson as the main build system, so we should stick to it. But I know others wants to have Makefile in the project to ease things for developers who are not that familiar with meson.

Anyway instead of `make rpm` I'd still leave the execution of `./build-scripts/build-rpm.sh` in the doc as the preferred solution ...

dougsland replied:
it's not about familiar to use meson, it's about keep both worlds for all devs as there is Makefile in the project. If the project want only use meson, just get out of Makefile would be appropriate IMO. I am going to update the patch and remove the `make rpm` thing for now.

engelmi committed:
Cleanup docs (#306)

* move hirte_e2e.md to the developer and test README

Signed-off-by: Michael Engel <mengel@redhat.com>

* re-add container base image file

The Containerfile-build had been deleted in
https://github.com/containers/hirte/pull/259

Signed-off-by: Michael Engel <mengel@redhat.com>

* restructured mkdocs

- moved and edited the dbus api description
- embedded the api examples for python
- moved and edited the proxy service md

Signed-off-by: Michael Engel <mengel@redhat.com>

---------

Signed-off-by: Michael Engel <mengel@redhat.com>

dougsland committed:
build-rpm: improve the bash script

Simplify the bash script, thanks to mperina.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

engelmi committed:
cleanup container files

- prune container files for integration tests
- pull base integration test container image
- added base image for github actions

Signed-off-by: Michael Engel <mengel@redhat.com>

engelmi committed:
update github workflows to use build-base image

Signed-off-by: Michael Engel <mengel@redhat.com>

pypingou committed:
Fix importing the python examples in the doc

Basically, our example files were not being found when the doc was built
on read-the-docs. This was due to the fact that rtd wasn't finding the
file. By specifying a base_path in the mkdocs.yml config file, all the paths to
the files we want to include can be made relative to this base_path.
Using this mechanism, rtd is able to find the snippets to include in the
doc and it is also working locally for developpers.

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Fix the links for editing the docs online

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Show the man pages in the docs

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Add a README explaining how to build the docs

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Another attempt to fix the edit links in the doc (#314)

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>
